### PR TITLE
add fieldselector

### PR DESCRIPTION
**Pull request by huoyinghao**
Signed-off-by: huoyinghao <962917302@qq.com>

### CODE REVIEW

File: fake/storage/memory/storage.go [hunk 1]
```diff
@@ -475,6 +484,47 @@ func (db *DataBase) LabelSelect(list []*Resource, s labels.Selector) ([]*Resourc
 	return res, nil
 }
 
+func (db *DataBase) EnhancedSelect(list []*Resource, s fields2.Selector) ([]*Resource, error) {
+	var res []*Resource
+	r, b := s.Requirements()
```
Comment (calvin0327) [inline fake/storage/memory/storage.go:489]: if r, b := s.Requirements(); b {
  for ...
}

File: fake/storage/memory/storage.go [hunk 2]
```diff
@@ -475,6 +484,47 @@ func (db *DataBase) LabelSelect(list []*Resource, s labels.Selector) ([]*Resourc
...
+	if b {
+		for _, obj := range list {
+			s2, err := obj.Object.MarshalJSON()
+			if err != nil {
+				return nil, err
+			}
+			var target map[string]interface{}
+			var ok bool
```
Comment (calvin0327) [inline fake/storage/memory/storage.go:497]: the `ok` seem a bit superfluous

**Comment by calvin0327** [comment]
@cleverhu PTAL

File: fake/storage/memory/storage.go [hunk 3]
```diff
@@ -475,6 +484,46 @@ func (db *DataBase) LabelSelect(list []*Resource, s labels.Selector) ([]*Resourc
...
+							return nil, err
+						}
+					} else {
+						n, ok := target[field].(string)
+						if !ok {
+							return nil, err
+						}
+						for _, value := range v.Values().List() {
+							if value == n {
+								res = append(res, obj)
```
Comment (cleverhu) [inline fake/storage/memory/storage.go:516]: This can return after append.Does't need to continue traversing.

Comment (cleverhu) [reply fake/storage/memory/storage.go:516]: Now it only supports equal matching, doesn't support not equal matching.

Comment (huoyinghao) [reply fake/storage/memory/storage.go:516]: builder.ListOptionsBuilder().Clusters("cluster01").Namespaces("default").FieldSelector("metadata.name",[]string{"pod01","pod02"}).Options()
这是 构造 field selector的 builder，构造出来之后 就是in,貌似 并没有not in 
(国人交流 如果可以说中文的话，真的赛高~~~)

**Pull request merged**